tools/generate-helpers/pg-table-bindings/main.go

```diff
@@ -193,46 +219,17 @@ func main() {
 	}
 }
 
-func generateSchema(s *walker.Schema, refs []*walker.Schema) error {
-	return generateSchemaRecursive(s, refs, set.NewStringSet(), schema.SchemaGenFS)
-}
-
-func generateSchemaRecursive(schema *walker.Schema, refs []*walker.Schema, visited set.StringSet, pkgPath string) error {
-	if !visited.Add(schema.Table) {
-		return nil
-	}
-
+func generateSchema(s *walker.Schema, searchCategory string, parsedReferences []parsedReference, dir string) error {
 	templateMap := map[string]interface{}{
-		"Schema":         schema,
-		"SearchCategory": "",
-		"Refs":           refs,
-	}
-
-	searchCategory, ok := typeToSearchCategoryMap[stringutils.GetAfter(schema.Type, ".")]
-	if ok {
-		templateMap["SearchCategory"] = fmt.Sprintf("v1.SearchCategory_%s", searchCategory)
+		"Schema":         s,
+		"SearchCategory": searchCategory,
+		"References":     parsedReferences,
 	}
 
-	if err := renderFile(templateMap, schemaTemplate, getSchemaFileName(pkgPath, schema.Table)); err != nil {
+	if err := renderFile(templateMap, schemaTemplate, getSchemaFileName(dir, s.Table)); err != nil {
 		return err
 	}
 
-	// No top-level schema has a parent unless attached synthetically.
-	for _, parent := range schema.ReferencedSchema {
```

Comment:
No, I'm just mandating that it eventually gets generated (which has to be the case).

For example, let's say schema A depends on schema B, and that schema A gets encountered by `go generate` first. The generated code for schema A _will_ reference by name the variable for schema B, even if it isn't generated yet. So the code won't compile in the time between when schema A is generated and when schema B is generated. But once schema B is generated, it's all fine.

(This is why I had to remove the import from `pg-table-bindings` to `central/postgres/schema`, btw. Otherwise, the fact that `central/postgres/schema` doesn't compile prevents us from running code generation... which is not ideal.)

Reply:
For example, I only generate namespace proto in one PR that references cluster proto. But since cluster is complex, it cannot be done. The compilation will fail because the current PR does not include cluster generation.
With new model, it kinda mandates refs be generated first, even if it must be putoff for later.

Reply:
Not really, you can generate the namespace tables but wait to add a reference to the cluster proto until the cluster code is generated. The reference is not meaningful anyway until there's a cluster SQL table.